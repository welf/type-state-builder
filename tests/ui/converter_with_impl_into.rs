use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct TestStruct {
    #[builder(converter = |value: String| value.to_uppercase(), impl_into)]
    name: String,
}

fn main() {}
