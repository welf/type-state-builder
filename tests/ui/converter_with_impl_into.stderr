error: Field-level converter is incompatible with impl_into
       note: #[builder(converter)] and #[builder(impl_into)] are incompatible
       help: use either custom converter or impl_into, not both
 --> tests/ui/converter_with_impl_into.rs:3:10
  |
3 | #[derive(TypeStateBuilder)]
  |          ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `TypeStateBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
